src/representative/rep_periods.jl

```diff
@@ -0,0 +1,200 @@
+
+"""
+    AbstractRepresentativePeriod{T} <: TimeStructure{T}
+
+Abstract type used for time structures that represent a representative period.
+These periods is obtained when iterating through the representative periods of a time
+structure declared using the function [`repr_period`](@ref.)
+"""
+abstract type AbstractRepresentativePeriod{T} <: TimeStructure{T} end
+
+function _rper(rp::AbstractRepresentativePeriod)
+    return error("_rper() not implemented for $(typeof(rp))")
+end
+isfirst(rp::AbstractRepresentativePeriod) = _rper(rp) == 1
+mult_repr(rp::AbstractRepresentativePeriod) = 1
+
+function Base.isless(
+    rp1::AbstractRepresentativePeriod,
+    rp2::AbstractRepresentativePeriod,
+)
+    return _rper(rp1) < _rper(rp2)
+end
+
+abstract type RepresentativeIndexable end
+
+struct HasReprIndex <: RepresentativeIndexable end
+struct NoReprIndex <: RepresentativeIndexable end
+
+RepresentativeIndexable(::Type) = NoReprIndex()
+RepresentativeIndexable(::Type{<:AbstractRepresentativePeriod}) = HasReprIndex()
+RepresentativeIndexable(::Type{<:TimePeriod}) = HasReprIndex()
+
+"""
+    SingleReprPeriod{T,OP<:TimeStructure{T}} <: AbstractRepresentativePeriod{T}
+
+A type representing a single representative period supporting iteration over its
+time periods. It is created through iterating through [`SingleReprPeriodWrapper`](@ref).
+"""
+struct SingleReprPeriod{T,OP<:TimeStructure{T}} <:
+       AbstractRepresentativePeriod{T}
+    ts::OP
+end
+
+_rper(rp::SingleReprPeriod) = 1
+
+mult_repr(rp::SingleReprPeriod) = 1.0
+
+Base.length(rp::SingleReprPeriod) = length(rp.ts)
+Base.eltype(::Type{SingleReprPeriod{T,OP}}) where {T,OP} = eltype(OP)
+
+StrategicIndexable(::Type{<:SingleReprPeriod}) = HasStratIndex()
+
+function Base.iterate(rp::SingleReprPeriod, state = nothing)
+    next = isnothing(state) ? iterate(rp.ts) : iterate(rp.ts, state)
+    return next
+end
+Base.last(rp::SingleReprPeriod) = last(rp.ts)
+
+"""
+    SingleReprPeriodWrapper{T,OP<:TimeStructure{T}} <: TimeStructure{T}
+
+Iterator for iterating through the individual presentative periods of a time structure
+without [`RepresentativePeriods`](@ref). It is automatically created through the function
+[`repr_periods`](@ref).
+"""
+struct SingleReprPeriodWrapper{T,OP<:TimeStructure{T}} <: TimeStructure{T}
+    ts::OP
+end
+
+"""
+    repr_periods(ts::TimeStructure)
+
+This function returns an iterator for iterating through the individual representative
+periods of a `TimeStructure`. The type of the iterator is dependent on the type of the
+input `TimeStructure`,
+
+When the `TimeStructure` is a `TimeStructure`, `repr_periods` returns a
```

Comment:
I would try to rephrase this a bit as the first sentence is kind of a tautology. I would also skip a reference to the `SinglePeriodReprWrapper` since this is documentation of an exported function. Maybe something along the lines of "If the time structure does not have representative periods, it will return the provided time structure as an iterable structure".

Reply:
Regarding the first point: I push an update. I guess iterator for iterating is a bit too much.

I am a bit skeptical regarding the second point. I would anyhow suggest to export the internal docstrings as well to an internal library as we do in EMX.